systemd/systemd.go

```diff
@@ -1065,18 +1082,53 @@ func (l Log) PID() string {
 	return "-"
 }
 
+type UnitLifetime int
+
+const (
+	Persistent UnitLifetime = iota
+	Transient
+)
+
 // MountUnitPath returns the path of a {,auto}mount unit
 func MountUnitPath(baseDir string) string {
 	escapedPath := EscapeUnitNamePath(baseDir)
 	return filepath.Join(dirs.SnapServicesDir, escapedPath+".mount")
 }
 
+// MountUnitPathWithLifetime returns the path of a {,auto}mount unit
+// created in the systemd directory suitable for the given unit lifetime
+func MountUnitPathWithLifetime(lifetime UnitLifetime, mountPointDir string) string {
+	escapedPath := EscapeUnitNamePath(mountPointDir)
+	var servicesPath string
+	switch lifetime {
+	case Persistent:
+		servicesPath = dirs.SnapServicesDir
+	case Transient:
+		servicesPath = dirs.SnapRuntimeServicesDir
+	default:
+		panic("unknown systemd unit lifetime")
```

Comment:
(nitpick) maybe this shold include the lifetime valid that it did not understand? i.e. `panin(fmt.Sprintf("unknown systemd unit lifetime %q", lifetime)`

Reply:
Done!